purchases/src/test/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsSynchronizerTest.kt

```diff
@@ -58,15 +59,49 @@ class DiagnosticsSynchronizerTest {
         )
     }
 
+    // region clearDiagnosticsFileIfTooBig
+
+    @Test
+    fun `clearDiagnosticsFileIfTooBig does nothing if dianostics file not too big`() {
+        every { diagnosticsFileHelper.isDiagnosticsFileTooBig() } returns false
+
+        diagnosticsSynchronizer.clearDiagnosticsFileIfTooBig()
+
+        verify(exactly = 0) { diagnosticsTracker.trackMaxEventsStoredLimitReached() }
+        verify(exactly = 0) { diagnosticsFileHelper.deleteDiagnosticsFile() }
+    }
+
+    @Test
+    fun `clearDiagnosticsFileIfTooBig tracks max events store limit reached if dianostics file too big`() {
+        every { diagnosticsFileHelper.isDiagnosticsFileTooBig() } returns true
+        every { diagnosticsTracker.trackMaxEventsStoredLimitReached() } just Runs
+
+        diagnosticsSynchronizer.clearDiagnosticsFileIfTooBig()
+
+        verify(exactly = 1) { diagnosticsTracker.trackMaxEventsStoredLimitReached() }
+    }
+
+    @Test
+    fun `clearDiagnosticsFileIfTooBig clears diagnostics file if dianostics file too big`() {
```

Comment:
What do you think about a more "functional" test that adds say 500 events and verifies that it's not considered too big?
As a way to validate that the size limit allows for some number of events that we're happy with.

Reply:
Good idea, done in https://github.com/RevenueCat/purchases-android/pull/1194/commits/265d1f66dc9f1b65d96f569b1c379d8d6ed88921